### PR TITLE
Swift Package Manager integration

### PR DESCRIPTION
**Pull request by immortal79**
Hi !

This PR enables us to import this wonderful framework directly within Xcode's built-in swift package manager.
I don't know if it's the right way to do it as it's the first time I do this. I Had to wrap every single class with #if canImport(UIKit) as SPM doesn't support compiling only for one platform as written here : [https://developer.apple.com/documentation/xcode/creating_a_standalone_swift_package_with_xcode](url)

I also did a PR for the dependency 'TransitionPatch' but you will need to manually change the URL to your repo in the Package.swift

Signed-off-by: Philippe Weidmann <philweidmann@me.com>

### CODE REVIEW

**Comment by muukii** [comment]
@immortal79 
Thank you for your contribution!
We'll be glad to review this.

Firstly, I think I have to make TransitionPatch lib supports SwiftPM.
https://github.com/muukii/TransitionPatch/pull/1
I'll check that.

File: Package.swift [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+let package = Package(
+    name: "Rideau",
+    platforms: [.iOS(.v10)],
+    products: [
+        .library(
+            name: "Rideau",
+            targets: ["Rideau"]),
+    ],
+    dependencies: [
+        .package(url: "https://github.com/immortal79/TransitionPatch.git", from: "1.0.3")
```
Comment (muukii) [inline Package.swift]: I created 1.0.3 as well.
https://github.com/muukii/TransitionPatch/releases/tag/1.0.3

Could you change the url to my repo's url?

```suggestion
        .package(url: "https://github.com/muukii/TransitionPatch.git", from: "1.0.3")
```

Comment (immortal79) [reply Package.swift]: Done 👍

**Comment by muukii** [comment]
@immortal79 
Thanks
And I've added to several checks with GitHub Actions on master-branch.
To check SwiftPM and CocoaPods capabilities, could you do rebase on master-branch?

**Comment by immortal79** [comment]
Sorry, I don't understand what you want me to do (I'm a basic git user) ?

**Comment by muukii** [comment]
@immortal79 
I got you, no problem.
I'd like this branch to contain the changes on master-branch to enable the checking (validating SwiftPM and CocoaPods).
In order to do, you need to `merge` or `rebase` with master-branch.
But, rebase might be a bit confusing.

So, It's no problem to use `merge`.
Could you try the following command?

In the current branch, and origin to your repo.

```swift
$ git remote add muukii git@github.com:muukii/Rideau
$ git fetch muukii
$ git merge muukii/master
$ git push origin master
```

**Comment by immortal79** [comment]
Thanks for the step by step instructions, I had to force push, hope it's okay.

**Comment by muukii** [comment]
@immortal79 
Thanks
but, it seems something went wrong.
Let me do this on the working branch.

**Comment by muukii** [comment]
https://github.com/muukii/Rideau/pull/27

**Pull request merged**